engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java

```diff
@@ -188,10 +243,115 @@ private void recursivelyAddModuleDependencies(Set<Name> modules, Name moduleName
         }
     }
 
+    /**
+     * Called whenever the user decides to add a new world.
+     * @param worldGeneratorInfo The {@link WorldGeneratorInfo} object for the new world.
+     */
     private void addNewWorld(WorldGeneratorInfo worldGeneratorInfo) {
         selectedWorld = worldGeneratorInfo.getDisplayName() + '-' + worldNumber;
-        worlds.put(selectedWorld, worldGeneratorInfo);
+        worlds.add(new WorldSetupWrapper(new Name(worldGeneratorInfo.getDisplayName() + '-' + worldNumber), worldGeneratorInfo));
         worldNumber++;
     }
+
+    /**
+     * This method switched the environment of the game to a temporary one needed for
```

Comment:
switched -> switches
created -> created

We write the javadocs in simple present, not past perfect.